jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/ProxyProvider.java

```diff
@@ -18,33 +18,58 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
 
 /** Propagates proxy configuration from Maven settings to system properties. */
 class ProxyProvider {
 
   private static final ImmutableList<String> PROXY_PROPERTIES =
       ImmutableList.of("proxyHost", "proxyPort", "proxyUser", "proxyPassword");
 
+  private static final ImmutableList<String> PROXY_PROTOCOLS = ImmutableList.of("https", "http");
   /**
    * Initializes proxy settings based on Maven settings.
    *
    * @param settings Maven settings
    */
-  static void init(DecryptedMavenSettings settings) {
-    configureProxy(settings, "https");
-    configureProxy(settings, "http");
-  }
+  static void populateSystemProxyProperties(Settings settings, SettingsDecrypter decrypter)
+      throws MojoExecutionException {
+    List<Proxy> proxies =
+        settings
+            .getProxies()
+            .stream()
+            .filter(Proxy::isActive)
+            .filter(proxy -> PROXY_PROTOCOLS.contains(proxy.getProtocol()))
+            .collect(Collectors.toList());
 
-  private static void configureProxy(DecryptedMavenSettings settings, String protocol) {
-    settings
-        .getProxies()
-        .stream()
-        .filter(Proxy::isActive)
-        .filter(proxy -> protocol.equals(proxy.getProtocol()))
-        .findFirst()
-        .ifPresent(ProxyProvider::setProxyProperties);
+    System.out.println(proxies.size());
+    if (proxies.size() == 0) {
+      return;
+    }
+
+    SettingsDecryptionRequest request = new DefaultSettingsDecryptionRequest().setProxies(proxies);
```

Comment:
Is it not possible to decrypt only the proxies we will use?

Reply:
Looks like I can fix this and [this](https://github.com/GoogleContainerTools/jib/pull/1712/files#r283058510) together.

Reply:
I was confused why the tests were passing, but we already have some code to prevent this (incase the user has set outside of the jib process), either way I'll reduce the search scope: https://github.com/GoogleContainerTools/jib/blob/a132b7f0a3a104d39875023deda60f125cb7e6e6/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/ProxyProvider.java#L58